packages/x-data-grid/package.json

```diff
@@ -52,7 +52,8 @@
     "@mui/x-internals": "workspace:*",
     "clsx": "^2.1.1",
     "prop-types": "^15.8.1",
-    "reselect": "^5.1.1"
+    "reselect": "^5.1.1",
+    "use-sync-external-store": "^1.0.0"
```

Comment:
Sidenote: if we're adding this dependency now, we could replace our `Store`/`useGridSelector` implementation with it. I don't remember why I didn't use it back then.